### PR TITLE
tpm2_getcap: support vendor properties

### PR DESCRIPTION
**Pull request by williamcroberts**
Requires tpm2-tss 4.0

Support vendor properties as a vendor agnostic TPM2B buffer. This also introduces the -i or --ignore-more option to ignore the moreData byte returned from the TPM.

Introduce -i to ignore moreData when it's set to 1 but moreData does not follow.

Example output from my NUC:
./tools/tpm2 getcap -i --tcti=pcap:device:/dev/tpmrm0 vendor:1 0000000100000003

Signed-off-by: William Roberts <william.c.roberts@intel.com>

### CODE REVIEW

**Comment by williamcroberts** [comment]
We should build a TCTI that takes data from stdin and writes to stdout so we can echo in a fake response buffer. I wonder if we can do this in bash and use the subprocess TCTI.

File: tools/tpm2_getcap.c [hunk 1]
```diff
@@ -801,6 +834,9 @@ static bool on_option(char key, char *value) {
     switch (key) {
     case 'l':
         options.list = true;
+        break;
+    case 'i':
+        options.ignore_more = true;
```
Comment (idesai) [inline tools/tpm2_getcap.c:839]: Elsewhere in all the tools `-i` is regarded as input. So maybe drop the short option.

File: tools/tpm2_getcap.c [hunk 2]
```diff
@@ -820,11 +856,12 @@ static bool on_arg(int argc, char *argv[]) {
 
 static bool tpm2_tool_onstart(tpm2_options **opts) {
 
-    const struct option topts[] = { { "list", no_argument, NULL, 'l' },
-
+    const struct option topts[] = {
+        { "list",        no_argument, NULL, 'l' },
+        { "ignore-more", no_argument, NULL, 'i' },
```
Comment (idesai) [inline tools/tpm2_getcap.c:861]:
```suggestion
        { "ignore-moredata", no_argument, NULL, 'i' },
```

File: tools/tpm2_getcap.c (hunk 1 above)
Comment (williamcroberts) [reply tools/tpm2_getcap.c:839]: I felt dirty about using -i, but it didn't even occur to me to just keep the long option.

**Comment by williamcroberts** [comment]
Nooooo, I forgot to submit the test tcti and now I am afraid I lost it :-(

**Comment by williamcroberts** [comment]
> Nooooo, I forgot to submit the test tcti and now I am afraid I lost it :-(

Thank you eclipse local history feature

**Comment by williamcroberts** [comment]
Pulling becuase tpm2-pytss CI is down.

**Pull request merged**